reana/reana_dev/git.py

```diff
@@ -84,6 +87,50 @@ def get_current_commit(srcdir):
     )
 
 
+def git_is_current_version_tagged(component):
+    """Determine whether the current version in source code is present as a git tag."""
+    tags_history = run_command(
+        "git tag --list", component, display=False, return_output=True
+    )
+    current_version = get_current_component_version_from_source_files(component)
+    return current_version in tags_history if tags_history else False
```

Comment:
This check may be dangerous, before the command returns one big string:

```
In [5]: run_command("git tag --list", "reana-commons", display=False, return_output=True)
Out[5]: 'v0.3.0\nv0.3.1\nv0.4.0\nv0.5.0\nv0.6.0\nv0.7.0a1\nv0.7.0a2\nv0.7.0a3'
```

so in theory one could get mismatches via matching substrings.

Perhaps change the implementation to check for a tag existence via `git tag -l <value>` technique.  E.g. compare:

```console
$ git tag -l 0.3.0 | wc -l
0
$ git tag -l v0.3.0 | wc -l
1
```